src/types/fhir-objects.d.ts

```diff
@@ -0,0 +1,22 @@
+declare module 'fhir-objects' {
+  import { R4 } from '@ahryman40k/ts-fhir-types';
+
+  type DomainResourceR4 = R4.IDomainResource & { resourceType: string };
+  type ResourceR4 = R4.IResource & { resourceType: string };
+  type DocumentReferenceR4 = R4.IDocumentReference & { status: string };
+
+  export type Address = fhir.Address | R4.IAddress;
+  export type DomainResource = fhir.DomainResource | DomainResourceR4;
+  export type DocumentReference = fhir.DocumentReference | DocumentReferenceR4;
+  export type HumanName = fhir.HumanName | R4.IHumanName;
+  export type Identifier = fhir.Identifier | R4.IIdentifier;
+  export type Observation = fhir.Observation | R4.IObservation;
+  export type Patient = fhir.Patient | R4.IPatient;
+  export type Practitioner = fhir.Practitioner | R4.IPractitioner;
+  export type Procedure = fhir.Procedure | R4.IProcedure;
+  export type Resource = fhir.Resource | ResourceR4;
+  export type Bundle = fhir.Bundle | R4.IBundle;
+  export type ServiceRequest = fhir.ProcedureRequest | R4.IServiceRequest;
+  export type MedicationRequest = fhir.MedicationRequest | R4.IMedicationRequest;
+  export type Request = ServiceRequest | MedicationRequest;
```

Comment:
Since this Request type is our own use-case based aggregation of otherwise independent types, can we somehow identify it as such? Not a huge deal but I would consider either renaming this, moving this elsewhere, or not making it a formal "type" at all.